### DISCUSSION TITLE
Operator log dir If too many directories are created, check whether the directory can be automatically cleared

### DISCUSSION THREAD

**Discussion by winfys** [apache/kyuubi#2152] [Q&A]
![image](https://user-images.githubusercontent.com/13195083/158555797-3bdced1c-b53c-420a-842f-63eb36bbf0ae.png)

**Comment by cxzl25** [comment]
At present, the log file of the current operation will be cleaned up during operation close, but the directory created by the session will not be cleaned up by session close, which should be a bug.

**Reply by cxzl25** [reply]
org.apache.kyuubi.operation.log.OperationLog#close

```scala
org.apache.kyuubi.operation.log.OperationLog#close
  def close(): Unit = synchronized {
    try {
      closeExtraReaders()
      reader.close()
      writer.close()
      Files.delete(path)

```

**Reply by yaooqinn** [reply]
seems reasonable

**Reply by cxzl25** [reply]
@winfys  do you want to try to fix this?

I fixed it locally and the validation works.

**Reply by winfys** [reply]
Yeah, I want to fix it
